apps/frontend/src/pages/access-management/components/group-users-table/group-users-table.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect } from "react";
+import { type FieldErrors, type UseFormSetValue } from "react-hook-form";
+
+import { Loader } from "~/libs/components/components.js";
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	usePagination,
+} from "~/libs/hooks/hooks.js";
+import {
+	actions as groupActions,
+	type GroupCreateRequestDto,
+} from "~/modules/groups/groups.js";
+
+import { UsersTable } from "../users-table/users-table.js";
+import { getRowId } from "./libs/helpers/helpers.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	errors: FieldErrors<GroupCreateRequestDto>;
+	onToggle: (id: number) => void;
+	selectedUserIds: number[];
+	setValue: UseFormSetValue<GroupCreateRequestDto>;
+};
+
+const GroupUsersTable = ({
+	errors,
+	onToggle,
+	selectedUserIds,
+	setValue,
+}: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+
+	const { users, usersDataStatus, usersTotalCount } = useAppSelector(
+		({ groups }) => groups,
+	);
+
+	const { onPageChange, onPageSizeChange, page, pageSize } = usePagination({
+		queryParameterPrefix: "user",
+		totalItemsCount: usersTotalCount,
+	});
+
+	useEffect(() => {
+		void dispatch(groupActions.configureGroupUsers({ page, pageSize }));
+	}, [dispatch, page, pageSize]);
+
+	useEffect(() => {
+		setValue("userIds", selectedUserIds);
+	}, [selectedUserIds, setValue]);
+
+	if (
+		usersDataStatus === DataStatus.IDLE ||
+		usersDataStatus === DataStatus.PENDING
+	) {
+		return <Loader />;
+	}
+
+	return (
+		<>
+			<span className={styles["table-title"]}>Users</span>
+			<UsersTable
+				getRowId={getRowId}
+				name="userIds"
+				onPageChange={onPageChange}
+				onPageSizeChange={onPageSizeChange}
+				onRowSelect={onToggle}
+				page={page}
+				pageSize={pageSize}
+				paginationBackground="secondary"
+				selectedIds={selectedUserIds}
+				totalItemsCount={usersTotalCount}
+				users={users}
+			/>
+			{selectedUserIds.length === EMPTY_LENGTH && errors.userIds && (
```

Comment:
why not to check for error message only? (the same logic as for inputs)
```tsx
	const error = errors["userIds"]?.message;
	const hasError = Boolean(error);
```
```suggestion
			{hasError && (
```

Reply:
Errors are refreshed on submit of the form and If we keep track of selectedUserIds.length it allows to not display error message when after unsuccessful form submit we select some user from the table. I thought it will be more responsible this way.

Reply:
Can't we use something more appropriate for this? for example reset method from react hook form

Reply:
I can use clearErrors from react hook form like this
```
	useEffect(() => {
		if (errors["userIds"] && selectedUserIds.length > EMPTY_LENGTH) {
			handleErrorsClear("userIds")
		}
	}, [selectedUserIds, errors, handleErrorsClear]);
```

Reply:
I've tried reset but it requires another method to get values and resetField just isn't resetting errors